force-app/main/default/classes/OpportunityRestResource.cls

```diff
@@ -0,0 +1,20 @@
+@RestResource(urlMapping='/services/apexrest/CompanyData/Opportunity/*')
```

Comment:
"/services/apexrest/" is adding to the url by default, so need to include it there as it'd be repeated twice in endpoint.
like "salesforce-instance/services/apexrest/services/apexrest/CompanyData/Opportunity/*"